boards: arm: add support for WeAct STM32G030 Core Board

heronet opened the pull request:
This patch adds support for the WeAct Studio STM32G030 Core Board, a compact and low-cost development platform based on the STM32G030F6P6 microcontroller.

Main features:
- STM32G030F6P6 MCU (32 KB Flash, 8 KB SRAM)
- User LED
- Exposed SWD header for programming/debugging
- UART1 (PB6/PB7) as default console
- I2C2 (PA11/PA12) and SPI1 (PA1/PA6/PA7) peripheral mapping
- 3.3V or 5V input power options with onboard regulator

The board has been tested with the following Zephyr samples:
- samples/basic/blinky
- samples/hello_world

nordicjm commented on boards/weact/stm32g030_core/weact_stm32g030_core_defconfig:
```diff
@@ -0,0 +1,16 @@
+# Kernel Options due to Low Memory (8k)
+CONFIG_MAIN_STACK_SIZE=640
+CONFIG_IDLE_STACK_SIZE=200
+CONFIG_ISR_STACK_SIZE=512
```
move these to a `Kconfig.defconfig` file

erwango replied:
See https://github.com/zephyrproject-rtos/zephyr/pull/92825

heronet replied:
Fixed. Thanks for the example

erwango commented on boards/weact/stm32g030_core/Kconfig.defconfig:
```diff
@@ -0,0 +1,22 @@
+	default 200 if SOC_STM32G030XX
+
+config ISR_STACK_SIZE
+	default 512 if SOC_STM32G030XX
```
Those `if SOC_STM32G030XX` should be removed as there is no doubt on this board.

erwango reviewed (changes requested):
@djiatsaf-st To be considered with your other PR

erwango reviewed (changes requested):
Otherwise LGTM

heronet commented:
A check failed due to API rate limit exceeding. i couldn't find a way to re-run the tests so I force pushed again.

erwango reviewed (dismissed):
To be merged after https://github.com/zephyrproject-rtos/zephyr/pull/92825

heronet commented on boards/weact/stm32g030_core/weact_stm32g030_core_defconfig:
```diff
@@ -0,0 +1,17 @@
+# Kernel Options due to Low Memory (8k)
+CONFIG_MAIN_STACK_SIZE=640
+CONFIG_IDLE_STACK_SIZE=200
+CONFIG_ISR_STACK_SIZE=512
```
So how is this going to be now? Kconfig.defconfig isn't ready yet for this

heronet replied:
Great. They are already removed.

erwango commented on boards/weact/stm32g030_core/Kconfig.defconfig:
```diff
@@ -0,0 +1,22 @@
+	default 200
+
+config ISR_STACK_SIZE
+	default 512
```
> I added the configs back in weact_stm32g030_core_defconfig

This is a useless burden.
Would have been easier to just remove the things, get approval and wait. This PR won't be merged before #92825

heronet replied:
Removed both. I'll add them once #92825 is merged

erwango replied:
> I'll add them once https://github.com/zephyrproject-rtos/zephyr/pull/92825 is merged

You'll have nothing to do, this will be handled at SoC level and not a board level anymore. We'll be able to merge this PR as is.

heronet committed:
boards: arm: add support for WeAct STM32G030 Core Board

This patch adds support for the WeAct Studio STM32G030 Core Board,
a compact and low-cost development platform based on the STM32G030F6P6
microcontroller.

Main features:
- STM32G030F6P6 MCU (32 KB Flash, 8 KB SRAM)
- User LED
- Exposed SWD header for programming/debugging
- UART1 (PB6/PB7) as default console
- I2C2 (PA11/PA12) and SPI1 (PA1/PA6/PA7) peripheral mapping
- 3.3V or 5V input power options with onboard regulator

The board has been tested with the following Zephyr samples:
- samples/basic/blinky
- samples/hello_world

Signed-off-by: Siratul Islam <sirat4757@gmail.com>
